.github/workflows/scancode.yml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# reusable workflow to call scancode action to scan licenses and copyrights and
+# upload the results.
+# TODO: add more usage guide here.
+name: 'scancode'
+on:
+  workflow_call:
+    inputs:
+      branch:
```

Comment:
This is for user to select a different branch to scan, for example I want to scan all the dependencies too, I can 'go mod vendor' (which download all vendor pkgs in the repo), then push it to a remote branch. On the GitHub UI, I can run the workflow to scancode the remote branch instead of main.